_test/general.test.php

```diff
@@ -58,4 +78,408 @@ public function test_plugin_conf() {
+     */
+    public function test_call_export_with_start_value_two() {
+        $pageId = 'demopage2';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(2, 1));
+    }
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = 0, count = 0, total = 1;
+     * @expectedException InvalidArgumentException
+     */
+    public function test_call_export_with_count_value_zero() {
+        $pageId = 'demopage3';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(0, 0));
+    }
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = 1, count = 0, total = 1;
+     * @expectedException InvalidArgumentException
+     */
+    public function test_call_export_with_count_value_zero_and_start_one() {
+        $pageId = 'demopage4';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(1, 0));
+    }
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = 0, count = -1, total = 1;
+     * @expectedException InvalidArgumentException
+     */
+    public function test_call_export_with_count_value_below_zero() {
+        $pageId = 'demopage5';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(0, -1));
+    }
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = -1, count = 1, total = 1;
+     * @expectedException InvalidArgumentException
+     */
+    public function test_call_export_with_start_value_below_zero() {
+        $pageId = 'demopage6';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(-1, 0));
+    }
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = 0, count = 10, total = 9;
+     */
+    public function test_call_export_with_count_greater_than_total() {
+        $pageId = array();
+        for ($i=1; $i<=9; $i++){
```

Comment:
Please run this through PHPStorm's formatter.

Reply:
This and everything :)

Reply:
I just formatted errything in the tests.
Forgot that :sweat_smile: .